src/PostgREST/Config.hs

```diff
@@ -282,6 +287,7 @@ parser optPath env dbSettings roleSettings roleIsolationLvl =
     <*> (fromMaybe False <$> optBool "server-timing-enabled")
     <*> (fmap T.unpack <$> optString "server-unix-socket")
     <*> parseSocketFileMode "server-unix-socket-mode"
+    <*> optString "admin-server-host"
```

Comment:
I think we could treat `configAdminServerHost` as a `Text` directly (without `Maybe`) and fallback to the default value at the moment of parsing the Config. This should avoid the extra `fromMaybe` done in other parts of the code.

You could use something like:

```suggestion
    <*> parseAdminServerHost "admin-server-host" "server-host"
```

And the implementation would be:

```haskell
    -- This should be replaced above: <*> parseServerHost "server-host"
    parseServerHost :: C.Key -> C.Parser C.Config Text
    parseServerHost key =
      fromMaybe "!4" <$> optString key

    parseAdminServerHost :: C.Key -> C.Key -> C.Parser C.Config Text
    parseAdminServerHost mainKey fallbackKey =
      optString mainKey >>= \case
        Just ash -> pure ash
        Nothing  -> parseServerHost fallbackKey
```

Reply:
That seems a very good approach, actually. We could get rid of other maybes in AppState this way as well.